taker-frontend/src/components/CloseButton.tsx

```diff
@@ -0,0 +1,79 @@
+import {
+    Box,
+    Button,
+    Popover,
+    PopoverArrow,
+    PopoverBody,
+    PopoverCloseButton,
+    PopoverContent,
+    PopoverFooter,
+    PopoverHeader,
+    PopoverTrigger,
+    Text,
+} from "@chakra-ui/react";
+import * as React from "react";
+import { Cfd, StateGroupKey, StateKey } from "../types";
+
+interface CloseButtonProps {
+    cfd: Cfd;
+    action: (req: any) => void;
+    status: boolean;
+    buttonName: String;
+}
+
+/// Button that can be used to trigger non-collaborative close if the maker is offline and collaborative close if the
+/// maker is online.
+export const CloseButton = ({ cfd, action, status, buttonName }: CloseButtonProps) => {
```

Comment:
I think we should prefer the format of:

```tsx
export default function CloseButton({ cfd, action, status, buttonName }: CloseButtonProps) {

}
```

1. If there is only one component in the file, the default export should be that component
2. `function`s are declarations and can thus be `default` exported, `const`s are declarations and cannot
```